### PR TITLE
prototype: Add ATB parameter

### PR DESCRIPTION
**Pull request by mrshu**
* Add ATB parameter to requests to DDG.

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

File: lib/ddg-atb.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                                             ["blocking"]);
+
+  function redirect(e) {
+    // Only change the URL if there is no ATB param specified.
+    if (e.url.indexOf('atb=') !== -1) {
+      return;
+    }
+
+    var newURL = e.url + "&atb=" + ss.storage.atb;
+    console.log("Redirecting: " + e.url + " to " + newURL);
```
Comment (bsstoner) [inline lib/ddg-atb.js]: Do you want to remove this console.log statement before merging?

Comment (mrshu) [inline lib/ddg-atb.js]: @bsstoner sure

**Comment by mrshu** [comment]
@bsstoner ready for merge

File: lib/main.js [hunk 2]
```diff
@@ -105,6 +106,14 @@ exports.main = function(options, callbacks) {
         ss.storage.meanings = true;
     }
 
+    // only set the atb parameter for new users
+    if (options.loadReason == "install") {
```
Comment (mrshu) [inline lib/main.js]: @bsstoner this makes sure only new installs get atb param set

File: lib/main.js [hunk 3]
```diff
@@ -169,10 +178,17 @@ exports.main = function(options, callbacks) {
...
         }, 1000);
     }
+
     pageMod.PageMod({
       include: /^https?:\/\/(www|encrypted)\.google\..*\/.*$/,
       contentStyleFile: './css/nopopup.css',
     });
+
+    // only redirect if there is an atb parameter set
+    if (ss.storage.atb != undefined) {
```
Comment (mrshu) [inline lib/main.js]: @bsstoner this makes sure that redirect happens only if there is an atb parameter set

File: lib/main.js (hunk 2 above)
Comment (bsstoner) [inline lib/main.js]: 👍

**Comment by bsstoner** [comment]
👍

**Pull request merged**